client/src/PagesBody/ContestsPage/ContestsPage.js

```diff
@@ -33,33 +36,47 @@ const ContestsPage = () => {
 
   useEffect(() => {
     async function fetchContestsData() {
-      const response = await fetch(
-        '/api/contests/card/data',
-        requestOptionsGET,
-      );
-      const result = await response.json();
+      let result = null;
+      if (
+        state.selectedRole === ROLE_NAME.PARTICIPANT &&
+        locationPath.state &&
+        locationPath.state.contestContent !== 'future'
+      ) {
+        result = await getContestsCards(ROLE_NAME.PARTICIPANT);
+      } else if (state.selectedRole === ROLE_NAME.MANAGER) {
+        result = await getContestsCards(ROLE_NAME.MANAGER);
+      } else if (state.selectedRole === ROLE_NAME.STAFF) {
+        result = await getContestsCards(ROLE_NAME.STAFF);
+      } else {
+        result = await getContestsCards();
+      }
+      if (Array.isArray(result)) {
+        result = removeNullsFromArray(result);
+      }
       rawDataFromDB.current = result;
+      setContestData(rawDataFromDB.current);
```

Comment:
They are the same, is there a reason for keeping both?

Reply:
changed! will be in new commit